examples/gamepad.html

```diff
@@ -0,0 +1,162 @@
+
+    button {
+      width: 120px;
+      height: 24px;
+      font-weight: bold;
+    }
+
+    select {
+      width: 120px;
+      font-size: 24px;
+    }
+  </style>
+</head>
+<body>
+  <h2>Select the sound source to play.</h2>
+  <h2>Use headphones for the full-sphere surround sound.</h2>
+  <p id="rotation">Please connect a gamepad (or a headtracker emulating a gamepad) to enable an 3D audio playback that considers your head rotations.</p>
+  <br>
+  <select id="eSelectSource" name="dropdown" disabled>
+    <option value="speech" selected="selected">Speech</option>
+    <option value="warbird">Airplanes</option>
+  </select>
+  <button id="eButtonPlayback" disabled>Play</button>
+
+  <script>
+    var audioSources = {
+      'speech': 'resources/4ch_B_FuMaNorm_FuMaOrd_speech.wav',
+      'warbird': 'resources/jlwarbirds3.amb'
+    };
+
+    var buttonPlayback, selectSource;
+    var yaw, pitch, roll;
+
+    window.addEventListener('load', function () {
+
+      // Set up the context and the audio element.
+      var audioContext = new AudioContext();
+      var audioElement = document.createElement('audio');
+      audioElement.loop = true;
+      audioElement.crossOrigin = 'anonymous';
+      audioElement.src = audioSources['speech'];
+
+      // Create a decoder.
+      var decoder = Omnitone.createFOADecoder(audioContext, audioElement, {
+        // The example audio is in the FuMa ordering (W,X,Y,Z). So remap the
+        // channels to the ACN format.
+        channelMap: [0, 3, 1, 2]
+      });
+
+      // Handles UI.
+      buttonPlayback = document.getElementById('eButtonPlayback');
+      selectSource = document.getElementById('eSelectSource');
+
+      buttonPlayback.onclick = function (event) {
+        if (event.target.textContent === 'Play') {
+          event.target.textContent = 'Pause';
+          audioElement.play();
+        } else {
+          event.target.textContent = 'Play';
+          audioElement.pause();
+        }
+      };
+
+      selectSource.onchange = function (event) {
+        audioElement.src = audioSources[event.target.value];
+        audioElement.load();
+        if (buttonPlayback.textContent === 'Pause')
+          audioElement.play();
+      };
+
+      var rotator;
+      
+      // Initialize the decoder and activate the button when ready.
+      decoder.initialize().then(function () {
+        buttonPlayback.disabled = false;
+        selectSource.disabled = false;
+        rotator = Omnitone.createFOARotator(audioContext);
+        window.requestAnimationFrame(runAnimation);
+      }, function (onInitializationError) {
+        console.error(onInitializationError);
+      });
+
+      var textRotation = document.getElementById('rotation');
+
+      function runAnimation() {
+
+        // look for a Gamepad and check, whether is has changed it axes values.
+        var gamepads = navigator.getGamepads ? navigator.getGamepads() : (navigator.webkitGetGamepads ? navigator.webkitGetGamepads() : []);
```

Comment:
Let's wrap the code at 80 col.

```js
var gamepads = navigator.getGamepads 
    ? navigator.getGamePads()
    : (navigator.webkitGetGamepads ? navigator.webkitGetGamepads : []);
```

What happens if `gamepads = []`?

Also does this have to be done every render call? Can we just cache the gamepad object upfront and use it?